### PR TITLE
Week 19 Happy Thoughts Api - Malin Göthe

### PR DESCRIPTION
**Pull request by M4L5**
Live project: https://malingoethe-happy-thoughts.netlify.app/
API project: https://goethe-happy-thoughts.herokuapp.com/
Will add readme and clean up after peer review :)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,28 +1,85 @@
...
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+import listEndpoints from 'express-list-endpoints'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndexes: true })
```
Comment (Sofarz) [inline server.js:10]: If I'm not mistaking it should be useCreateIndex and not indexes here.

File: server.js [hunk 2]
```diff
@@ -1,28 +1,85 @@
...
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.use((_, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service not available' })
+  }
+})
```
Comment (Sofarz) [inline server.js:45]: Nice! 💯

File: server.js [hunk 3]
```diff
@@ -1,28 +1,85 @@
...
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const newLike = await Thought.findByIdAndUpdate({ id }, {$inc: {hearts: 1} })
```
Comment (Sofarz) [inline server.js:73]: It is not possible to add likes right now, I checked it in Postman and it does not work there, so there is something wrong in the backend. I believe that you only need to write id on line 73 and not { id } :)

File: server.js [hunk 4]
```diff
@@ -1,28 +1,85 @@
...
+})
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', (_, res) => {
+  res.send(listEndpoints(app))
 })
 
+app.get('/thoughts', async (_, res) => {
```
Comment (Sofarz) [inline server.js:51]: Nice feature with the _ instead of req when it isn't used, but not know about that 👍

**Review by Sofarz — CHANGES_REQUESTED** [review]
Malin! 🌷 
Great work on this weeks project! However there are some things you need to fix to make it possible to add likes. Look at my comment regarding line 73 :).
A tip for the future would be to add more comments, to make it easier for yourself to follow if you go back to the code, as well as for others 😃 .

File: server.js [hunk 5]
```diff
@@ -1,28 +1,85 @@
...
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    trim: true,
+    minlength: 5,
+    maxlength: 140
```
Comment (KarolCislowski) [inline server.js:22]: Nicely configured field 👍🏻

File: server.js [hunk 6]
```diff
@@ -1,28 +1,85 @@
...
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', (_, res) => {
+  res.send(listEndpoints(app))
 })
 
+app.get('/thoughts', async (_, res) => {
+  try {
+    const everyThought = await Thought.find().sort({ createdAt: -1 }).limit(20)
```
Comment (KarolCislowski) [inline server.js:53]: 👍🏻

File: server.js [hunk 7]
```diff
@@ -1,28 +1,85 @@
...
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const newLike = await Thought.findByIdAndUpdate(id, {$inc: {hearts: 1} })
```
Comment (KarolCislowski) [inline server.js:73]: 👍🏻

**Review by KarolCislowski — APPROVED** [review]
Great API but you need to work on your frontend 👨🏻‍🎨 🎨

**Pull request closed without merging**